crates/pulsar-core/src/pdk/process_tracker.rs

```diff
@@ -349,6 +386,43 @@ impl ProcessTracker {
             }
         });
     }
+
+    /// Check if a PID is descendant of a target image
+    fn is_descendant_of(&self, pid: Pid, target_image: &str) -> Result<bool, TrackerError> {
+        let mut process = self.data.get(&pid).ok_or(TrackerError::ProcessNotFound)?;
+
+        // First, check if the process itself is the target image
+        if process.original_image.eq(target_image)
+            || process
+                .exec_changes
+                .values()
+                .any(|image| image.eq(target_image))
+        {
+            return Ok(true);
+        }
```

Comment:
This can be removed: the first iteration of the loop executes the very same code